core/templates/dev/head/pages/exploration_editor/statistics_tab/EarlyQuitIssueDirective.js

```diff
@@ -0,0 +1,149 @@
+        // An integer representing the issue index.
+        index: '&',
+        // A read-only object representing the issue.
+        issue: '&'
+      },
+      templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+        '/pages/exploration_editor/statistics_tab/' +
+        'early_quit_issue_directive.html'),
+      controller: [
+        '$scope', '$uibModal', 'IssuesService',
+        function(
+            $scope, $uibModal, IssuesService) {
+          $scope.currentIssueIndex = $scope.index();
+
+          var issue = $scope.issue();
+
+          $scope.issueStatement = IssuesService.renderIssueStatement(issue);
+
+          $scope.suggestions = IssuesService.renderIssueSuggestions(issue);
+
+          $scope.playthroughIds = issue.playthroughIds;
+
+          $scope.getPlaythroughIndex = function(playthroughId) {
+            return $scope.playthroughIds.indexOf(playthroughId);
+          };
+
+          $scope.showPlaythrough = function(playthroughId) {
+            IssuesService.getPlaythrough(
+              playthroughId
+            ).then(function(playthrough) {
+              $uibModal.open({
+                templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+                  '/pages/exploration_editor/statistics_tab/' +
+                  'playthrough_modal_directive.html'),
+                backdrop: true,
+                resolve: {
+                  playthrough: function() {
+                    return playthrough;
+                  },
+                  playthroughIndex: function() {
+                    return $scope.playthroughIds.indexOf(playthroughId);
+                  }
+                },
+                controller: [
+                  '$scope', '$uibModalInstance', 'playthroughIndex',
+                  'playthrough', 'AlertsService', 'LearnerActionRenderService',
+                  function(
+                      $scope, $uibModalInstance, playthroughIndex,
+                      playthrough, AlertsService, LearnerActionRenderService) {
+                    $scope.playthrough = playthrough;
+                    $scope.playthroughIndex = playthroughIndex;
+
+                    console.log($scope.playthrough.actions);
+                    $scope.displayBlocks =
+                      LearnerActionRenderService.getDisplayBlocks(
+                        playthrough.actions);
+
+                    var blockActionIndexMapping = {};
+                    var i = $scope.displayBlocks.length - 1, accumulator = 1;
+                    for (i; i >= 0; i--) {
+                      blockActionIndexMapping[i] = accumulator;
+                      accumulator += $scope.displayBlocks[i].length;
+                    }
+
+                    $scope.maxHidden = $scope.displayBlocks.length - 1;
+                    console.log($scope.maxHidden);
+
+                    $scope.getDisplayBlockIndex = function(displayBlock) {
+                      return $scope.displayBlocks.indexOf(displayBlock);
```

Comment:
Are you sure we need so many things in `$scope`? Try to minimize the things you make public there and ideally only use helper functions like these to display things.

Reply:
Done.